### ISSUE TITLE
connecting Iphython to jupyter notebook

### ISSUE THREAD

**Issue by mischki** [microsoft/vscode-jupyter#6484]
Is there a way to have an interactive command line that is connected to a jupyter notebook? Not a python script divided up into "jupyter cells", which works fine, but a notebook. Running 'Jupyter : Create Interactive Window" creates a shell that doesn't see the variables in Jupyter Notebook's workspace. This functionality was available in plain Jupyter , run in a browser, using the "Scratch Pad" extension. 
Thanks.

**Comment by IanMatthewHuff** [comment]
Thanks for the suggestion, I know that we've batted around the idea of a scratchpad extension a few times for VS Code. We'll put it in for consideration. Personally I think it would be an excellent addition.

**Comment by mischki** [comment]
Thank you. Are you aware of any workarounds?

**Comment by philffm** [comment]
> Thank you. Are you aware of any workarounds?

Hi @mischki have you found any workarounds in the meantime? Really slows down my trial–and-error development process I became so used to when using the VSCode python debugger.

**Comment by mischki** [comment]
> > Thank you. Are you aware of any workarounds?
> 
> Hi @mischki have you found any workarounds in the meantime :)? Really slows down my trial–and-error development process I became so used to when using the VSCode python debugger.
> 
> Cheers, Phil

Hi Phil,
I'm still using Jupyter in Chrome and anxiously googling about this every week. There seems to be no interest among the developers.

**Comment by philffm** [comment]
@mischki 

Cheers! Seems like I managed to got it working. (can use the Jupyter Notebook variables in the interactive window now. A bit of a process but imho worth it :D)

## The process

1) Command + Shift + P
2) Jupyter: Create Interactive Window
3) Copy URL of Remote Jupyter Session
4) again: Command + Shift + P -> Jupyter: Specify local or remote Jupyter server for connections
5) Paste URL of Remote Jupyter Session
6) In the interactive window select the current session (remote)
7) That's it! The active variables from the Jupyter Notebook should now available in the interactive console

**Comment by mischki** [comment]
Hi. 
Is your Jupyter Notebook opened within VSCode or in a browser? If in VSCode, how do you copy its url? 
Thanks.

**Comment by philffm** [comment]
@mischki haha, that's a bit tricky indeed: I copy the entire block, paste it in the execution field and from there I can easily copy the URL
<img width="689" alt="Screenshot 2022-01-22 at 18 13 37" src="https://user-images.githubusercontent.com/6079545/150648695-03e43c8e-b809-4cb6-a666-bb35589571ea.png">
<img width="663" alt="Screenshot 2022-01-22 at 18 13 05" src="https://user-images.githubusercontent.com/6079545/150648696-390018ef-abf2-415e-a325-3139e54b6138.png">

**Comment by mischki** [comment]
Sorry, I'm quite new to VSCode. How do you get to that window (first picture) from where you copy the URL? When I open a Jupyter notebook in VSCode I don't see a window with a URL like that. Thanks.

**Comment by philffm** [comment]
![image](https://user-images.githubusercontent.com/6079545/150672360-dd452713-dc95-4539-a46a-f33f6c76b651.png)
This one gets you to the interactive window. (Command - Shift - P (guess something similar when you're using Windows / Linux)) Hope that helps :)

**Comment by mischki** [comment]
> 

I meant this one 
![image](https://user-images.githubusercontent.com/31024913/150687270-3a52b064-d5cf-4ff6-8d1b-4100564e446c.png)

**Comment by mischki** [comment]
> > 
> 
> I meant this one ![image](https://user-images.githubusercontent.com/31024913/150687270-3a52b064-d5cf-4ff6-8d1b-4100564e446c.png)

When I start a Jupyter Interactive Window this is what I get
![image](https://user-images.githubusercontent.com/31024913/150857235-e43e26f0-4c74-4cd5-82bf-7fb99c41a7a3.png)
No URL anywhere. So I still don't know how to get the URL of the Notebook's kernel to paste into the interactive window.
What am I missing?
Thanks

**Comment by notuntoward** [comment]
Yes, I'm also really missing the console in Jupyterlab.  Here's a related vscode request too: [#100061](https://github.com/microsoft/vscode/issues/100061)

**Comment by philffm** [comment]
@mischki Now it also happened to me - there was no URL shown anymore in the March 2022 version of VSCode - so I downgraded to the Feb version and it started working again 🤷🏽‍♂️

**Comment by martincadek** [comment]
> @mischki Now it also happened to me - there was no URL shown anymore in the March 2022 version of VSCode - so I downgraded to the Feb version and it started working again 🤷🏽‍♂️

I tested the approach using terminal and typing: `jupyter notebook` as suggested [here](https://stackoverflow.com/questions/73583262/obtaining-localhost-url-of-python-kernel-for-jupyter-notebook-interactive-window). I can confirm that this works as well and ther's no need to downgrade or change VS Code versions.

_The process is similar:_

1. Open terminal and type `jupyter notebook` in terminal and copy localhost (it will also open jupyter in browser by default but you don't need to keep running it).
2. Command + Shift + P
3. Specify Jupyter Server for Connections
4. Paste copied URL
5. Command + Shift + P
6. Open interactive window and select the remote kernel you specified earlier
7. Select the same remote kernel also in your jupyter session, i.e. local notebook so identical kernel is in interactive and local session. Now you should have output carried from one your jupyter notebook to your interactive window.

**Comment by amunger** [comment]
duplicate of https://github.com/microsoft/vscode-jupyter/issues/4573
